src/tracks/Pattern.cpp

```diff
@@ -125,6 +125,55 @@ void Pattern::init()
 
 	changeLength( length() );
 	restoreJournallingState();
+
+
+	// Resize this track to be the same size as existing tracks in the BB
+
+	TrackContainer * currentTrackContainer = m_instrumentTrack->trackContainer();
+
+	// Only resize if there actually exists another track in the BB
+	if(currentTrackContainer->tracks().size() > 1)
+	{
+		// Do this per BB for the current Track
+		for(unsigned int tcoIndex = 0; tcoIndex < m_instrumentTrack->getTCOs().size();
+			++tcoIndex)
```

Comment:
This should be done only for the TCO the pattern belongs to. You can get the `tcoIndex` with `getTCOs().indexOf(this)`.